### DISCUSSION TITLE
Can I store SNMP string in Netbox ?

### DISCUSSION THREAD

**Discussion by Jason-MK** [netbox-community/netbox#7563] [Q&A]
Hello ,  

 I want to ask If I can store SNMP strings in Netbox so when I use an API call I can read it ? without the need of using the Tags ?    

Best Regards

**Comment by candlerb** [comment]
Do you care how "secret" these values are, e.g. if the cleartext values are visible in the database?  If not, then you can use Custom Fields against a Device or VM, or Config Contexts.

If they need to be encrypted, then there used to be a function called "[Secrets](https://github.com/netbox-community/netbox/tree/v2.11.12/docs/models/secrets)".  As of Netbox 3.0, this has been moved into a [plugin](https://github.com/DanSheps/netbox-secretstore).  These are also stored against a Device/VM, and you need to provide a user key to decrypt them.

However, I think that you'll find that a dedicated secret store like Hashicorp Vault is much more usable (especially to API consumers) and more secure-by-design.

Speaking for myself, I use the same community strings across many devices, so I don't want to store the same value multiple times against each device.  For these use cases, storing the secrets in Vault makes a lot more sense.  I suggest you use a tag or custom field in Netbox to refer to the *name* of the secret, and then fetch the required secret from Vault when required.

Vault has a number of very useful features, including [versioned secrets](https://learn.hashicorp.com/tutorials/vault/versioned-kv).

**Reply by mraerino** [reply]
see this discussion for a Vault plugin: https://github.com/netbox-community/netbox/discussions/7980
